interface/mmal/util/mmal_component_wrapper.c

```diff
@@ -151,7 +151,7 @@ MMAL_STATUS_T mmal_wrapper_create(MMAL_WRAPPER_T **ctx, const char *name)
    if (status != MMAL_SUCCESS)
       return status;
 
-   extra_size = (component->input_num + component->output_num) * 2;
+   extra_size = (component->input_num + component->output_num * 2) * sizeof(*private);
```

Comment:
I'll agree it is wrong as written, but it's not * sizeof(*private).
Looking at line 176 it's using each extra entry only as a pointer to either a MMAL_POOL_T, or a MMAL_QUEUE_T. There's a MMAL_POOL_T per input port, and both a MMAL_POOL_T and MMAL_QUEUE_T per output port. So I'd say something along the lines of 
```
extra_size = (component->input_num * sizeof(MMAL_POOL_T*)) + (component->output_num * (sizeof(MMAL_POOL_T*) + sizeof(MMAL_QUEUE_T*))
```
would be the desired value.

I've never used these wrapper functions (I'm not sure anyone did in anger), so there may well be gremlins lurking.

Reply:
You are missing the last `)`, but, yeah, your formula is better to understand (the resultant value is the same as mine since sizeof(*some pointer*) is always 4). I've pushed this change.

The change just increases a memory to allocate, so no problems are going to occur it seems.

Reply:
(I think I'm right on this, but could be hideously embarrassing myself!)

My main point was that you hadn't done ```sizeof(some pointer)``` though, you'd done ```sizeof(*(some pointer))```, ie the sizeof the thing pointed to, or ```sizeof(MMAL_WRAPPER_PRIVATE_T)```. That is != 4. (76 I think - I haven't got code to hand to invoke it).

```sizeof(private)``` I would have accepted (slightly grudgingly), as it is just a pointer and would be size 4, but having looked into what it is using that for then it's far better to be specific..

Reply:
Oh I missed it. Thanks!